src/Azure.Functions.Cli/Diagnostics/ColoredConsoleLogger.cs

```diff
@@ -5,50 +5,74 @@
 using Microsoft.Extensions.Logging;
 using Azure.Functions.Cli.Common;
 using static Azure.Functions.Cli.Common.OutputTheme;
+using System.Linq;
+using System.Diagnostics.Tracing;
 
 namespace Azure.Functions.Cli.Diagnostics
 {
     public class ColoredConsoleLogger : ILogger
     {
-        private readonly Func<string, LogLevel, bool> _filter;
         private readonly bool _verboseErrors;
         private readonly string _category;
+        private readonly LoggingFilterHelper _loggingFilterOptions;
+        private readonly string[] allowedLogsPrefixes = new string[] { "Worker process started and initialized.", "Host lock lease acquired by instance ID" };
 
-        public ColoredConsoleLogger(string category, Func<string, LogLevel, bool> filter = null)
+        public ColoredConsoleLogger(string category, LoggingFilterHelper loggingFilterOptions)
```

Comment:
nit: change the param name from `Options` to `Helper` (there's a few other places for this as well).